src/transformers/api-history.ts

```diff
@@ -0,0 +1,262 @@
+  changes?: { 'pr-url': string; description: string }[];
+};
+
+export interface PrReleaseVersions {
+  release: string | null;
+  backports: Array<string>;
+}
+export type PrReleaseVersionsContainer = { [key: number]: PrReleaseVersions };
+
+interface PrReleaseArtifact {
+  data: PrReleaseVersionsContainer;
+  endCursor: string;
+}
+
+export default function attacher() {
+  return transformer;
+}
+
+function matchApiHistoryCodeBlock(node: Node): node is Code {
+  return isCode(node) && node.lang === 'YAML' && node.meta === 'history';
+}
+
+let _allElectronVersions: SemVer[];
+
+async function getAllElectronVersions(): Promise<SemVer[]> {
+  if (_allElectronVersions) {
+    return _allElectronVersions;
+  }
+
+  // TODO: Error handling
+  const { versions } = await ElectronVersions.create(undefined, {
+    ignoreCache: true,
+  });
+  _allElectronVersions = versions;
+  return _allElectronVersions;
+}
+
+let _allPrReleaseVersions: PrReleaseVersionsContainer;
+
+async function getAllPrReleaseVersions(): Promise<PrReleaseVersionsContainer> {
+  if (_allPrReleaseVersions) {
+    return _allPrReleaseVersions;
+  }
+
+  // TODO: Add error handling and logging
+  if (process.env.GITHUB_ACTIONS === 'true' && !process.env.GH_TOKEN) {
+    throw new Error('GH_TOKEN is required when running in GitHub Actions.');
+  }
+
+  // TODO: Remove this
+  if (!process.env.GH_TOKEN) {
+    const versions: PrReleaseVersionsContainer = {
+      35658: {
+        release: 'v30.0.0-nightly.20231214',
+        backports: ['v29.0.0-alpha.9'],
+      },
+      41391: {
+        release: 'v31.0.0-alpha.1',
+        backports: [] as string[],
+      },
+      42086: {
+        release: 'v32.0.0-nightly.20240531',
+        backports: ['v31.0.0'],
+      },
+    };
+
+    _allPrReleaseVersions = versions;
+    return _allPrReleaseVersions;
+  }
+
+  // ? Maybe log this?
+  if (!process.env.GH_TOKEN) {
+    _allPrReleaseVersions = {};
+    return _allPrReleaseVersions;
+  }
+
+  const fetchOptions = {
+    method: 'GET',
+    headers: {
+      Accept: 'application/vnd.github+json',
+      'X-GitHub-Api-Version': '2022-11-28',
+      Authorization: `Bearer ${process.env.GH_TOKEN}`,
+    },
+  };
+
+  const artifactsListResponse = await fetch(
+    'https://api.github.com/repos/electron/website/actions/artifacts',
+    fetchOptions
+  );
+  const latestArtifact = (await artifactsListResponse.json()).artifacts
+    .filter(({ name }: { name: string }) => name === 'resolved-pr-versions')
+    .sort((a: { id: number }, b: { id: number }) => a.id > b.id)[0];
+
+  // ? Maybe use streams/workers
+  const archiveDownloadResponse = await fetch(
+    latestArtifact.archive_download_url,
+    fetchOptions
+  );
+  const buffer = Buffer.from(await archiveDownloadResponse.arrayBuffer());
+
+  const zip = new AdmZip(buffer);
+  const parsedData = JSON.parse(
+    zip.readAsText(zip.getEntries()[0]!)
+  ) as PrReleaseArtifact;
```

Comment:
> @erickzhao https://github.com/electron/lint-roller/pull/73#discussion_r1670619581
> nit: Would it be possible to expand this line into a few more to make it easier to debug if something goes wrong?
> 
> For example:
> 
> * Checking `Array.isArray` and `length` instead of non-null coercing
> * An `isPRReleaseArtifact` type guard function instead of a type assertion

Reply:
Resolved in [`394cc14` (#594)](https://github.com/electron/website/pull/594/commits/394cc144af0fd1c1938304be8ea982cfb5f56ede)